ta/pkcs11/src/processing_symm.c

```diff
@@ -751,6 +857,78 @@ enum pkcs11_rc step_symm_operation(struct pkcs11_session *session,
 
 		break;
 
+	case PKCS11_CKM_MD5_HMAC_GENERAL:
+	case PKCS11_CKM_SHA_1_HMAC_GENERAL:
+	case PKCS11_CKM_SHA224_HMAC_GENERAL:
+	case PKCS11_CKM_SHA256_HMAC_GENERAL:
+	case PKCS11_CKM_SHA384_HMAC_GENERAL:
+	case PKCS11_CKM_SHA512_HMAC_GENERAL:
+		assert(session->processing->extra_ctx);
+		hmac_len = *(uint32_t *)session->processing->extra_ctx;
+
+		switch (function) {
+		case PKCS11_FUNCTION_SIGN:
+			if (out_size && out_size < hmac_len) {
+				/* inform client of required size */
+				out_size = hmac_len;
+				return PKCS11_CKR_BUFFER_TOO_SMALL;
```

Comment:
Issue the output size not returned.
```patch
-				return PKCS11_CKR_BUFFER_TOO_SMALL;
+				output_data = true;
+				rc = PKCS11_CKR_BUFFER_TOO_SMALL;
+				goto out;

Reply:
Ok. With this change, I was able to simplify the code a bit as well. Please see if the following changes are acceptable. Thanks.

Reply:
LGTM